robot_framework/process.py

```diff
@@ -1,8 +1,343 @@
+        _send_status_email(_get_recipient_from_email(data_dict["Bruger"]), is_user_recognized, data_dict["Sagsoverskrift"])
+        # If user is not allowed to send this data, stop the process.
+        if not is_user_recognized:
+            continue
+        list_of_ids = _get_ids_from_mail(email, graph_access)
+        orchestrator_connection.bulk_create_queue_elements(
+            config.QUEUE_NAME,
+            references = list_of_ids,
+            data=[json.dumps(data_dict, ensure_ascii=False)] * len(list_of_ids),
+            created_by="Robot")
+        graph_mail.delete_email(email, graph_access)
+
+
+def _get_az_from_email(user_data: str) -> str:
+    '''Find az in user_data using regex'''
+    pattern = r"\baz\d+\b"
+    return re.findall(pattern, user_data)[0]
+
+
+def _get_recipient_from_email(user_data: str) -> str:
+    '''Find email in user_data using regex'''
+    pattern = r"E-mail: (\S+)"
+    return re.findall(pattern, user_data)[0]
+
+
+def _check_az(orchestrator_connection: OrchestratorConnection, email_az: str) -> bool:
+    '''Check AZ from email against accepted list of AZ in process arguments.
+
+    Args:
+        orchestrator_connection: Connection containing process arguments with some accepted AZs
+        email_az: A user identification (AZ) read from an email
+    '''
+    accepted_azs = json.loads(orchestrator_connection.process_arguments)["accepted_azs"]
+    return email_az.lower() in [az.lower() for az in accepted_azs]
+
+
+def _send_status_email(recipient: str, process_started: bool, case_name: str):
+    '''Send an email with variable text depending on whether the process started or not.
+
+    Args:
+        recipient: Who should receive the email
+        process_started: Did the process start?
+        case_name: Which case does this concern?
+    '''
+    subject = config.SUBJECT_BASE
+    text = config.BODY_BASE + case_name
+    if process_started:
+        subject += config.SUBJECT_SUCCESS
+        text += config.BODY_SUCCESS
+    else:
+        subject += config.SUBJECT_ERROR
+        text += config.BODY_ERROR
+    smtp_util.send_email(
+        receiver= recipient,
+        sender=config.STATUS_SENDER,
+        subject=subject,
+        body=text,
+        smtp_server=config.SMTP_SERVER,
+        smtp_port=config.SMTP_PORT
+    )
+
+
+def _create_notes_from_queue(orchestrator_connection: OrchestratorConnection, nova_access: NovaAccess):
+    ''' Load queue elements and write notes to KMD Nova
+
+    Args:
+        orchestrator_connection: A way to read the queue elements
+        nova_access: A token to write the notes
+    '''
+    queue_elements_processed = 0
+    while (queue_element := orchestrator_connection.get_next_queue_element(config.QUEUE_NAME)) and queue_elements_processed < config.MAX_TASK_COUNT:
+        # Find a case to add note to
+        data_dict = json.loads(queue_element.data)
+        cases = nova_cases.get_cases(nova_access, cpr = queue_element.reference)
```

Comment:
Why do you get the cases here?

Reply:
If we want to look up an existing case, we need to look at the existing cases, but I get how it's confusing. I've split the function both looking for existing cases and making a new case to two functions and made it clearer, I think.